libs/platform/experience/composition/composition-components.controller.ts

```diff
@@ -51,11 +58,13 @@ export class CompositionComponentsController implements ReactiveController {
         uid
           ? this.experienceService.getComponent({ uid }).pipe(
               switchMap((component) => {
-                const components = component?.components;
+                if (!component?.components) return of(null);
 
-                if (!components) {
-                  return of(null);
-                }
+                const components = Array.isArray(component.components)
+                  ? component.components
+                  : this.host.bucket
+                  ? component.components[this.host.bucket]
+                  : component.components.main;
```

Comment:
opinionated, but wondering what's more readable

```suggestion
                const components = Array.isArray(component.components)
                  ? component.components
                  : component.components[this.host.bucket ?? 'main']
```